platform_plugin_turnitin/turnitin.py

```diff
@@ -0,0 +1,426 @@
+
+    def create_turnitin_submission_object(self):
+        """
+        Constructs a Turnitin submission object based on the user's data.
+
+        Returns:
+            Response: The response from the Turnitin submission API.
+        """
+        user_data = self.get_user_data()
+        user_name = user_data["user_name"]
+        date_now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        payload = {
+            "owner": user_data["user_id"],
+            "title": self.location.block_id,
+            "submitter": user_data["user_id"],
+            "owner_default_permission_set": "LEARNER",
+            "submitter_default_permission_set": "INSTRUCTOR",
+            "extract_text_only": False,
+            "metadata": {
+                "owners": [
+                    {
+                        "id": user_data["user_id"],
+                        "given_name": user_name[0] if user_name else "no_name",
+                        "family_name": " ".join(user_name[1:])
+                        if len(user_name) > 1
+                        else "no_last_name",
+                        "email": user_data["user_email"],
+                    }
+                ],
+                "submitter": {
+                    "id": user_data["user_id"],
+                    "given_name": user_name[0] if user_name else "no_name",
+                    "family_name": " ".join(user_name[1:])
+                    if len(user_name) > 1
+                    else "no_last_name",
+                    "email": user_data["user_email"],
+                },
+                "original_submitted_time": date_now,
+            },
+        }
+        return post_create_submission(payload)
+
+    @XBlock.handler
+    def upload_turnitin_submission_file(self, data, suffix=""):
+        """
+        Handles the upload of the user's file to Turnitin.
+
+        Args:
+            data (WebRequest): Web request containing the file to be uploaded.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            Response: The response after uploading the file to Turnitin.
+        """
+        turnitin_submission = self.create_turnitin_submission_object()
+        if turnitin_submission.status_code == 201:
+            turnitin_submission_id = turnitin_submission.json()["id"]
+            current_user_id = self.get_user_data()["user_id"]
+            current_user = User.objects.get(id=current_user_id)
+            submission = TurnitinSubmission(
+                user=current_user, turnitin_submission_id=turnitin_submission_id
+            )
+            submission.save()
+            myfile = data.params["myfile"].file
+            # turnitin_submission_id='0a966646-83f9-4ce6-aa47-71e07baf4e30'
+            response = put_upload_submission_file_content(
+                turnitin_submission_id, myfile
+            )
+            return Response(
+                json.dumps(response.json()),
```

Comment:
if you do response.txt don't you already get the json format? Not sure though, you might need to check